src/fsharp/FSharp.Core/fslib-extra-pervasives.fs

```diff
@@ -70,6 +81,78 @@ module ExtraTopLevelOperators =
                     let key = RuntimeHelpers.StructBox(k)
                     if d.ContainsKey(key) then (r <- d.[key]; true) else false
                 member s.Remove(k : 'Key) = (raise (NotSupportedException(SR.GetString(SR.thisValueCannotBeMutated))) : bool) 
+#if FX_NO_READONLY_COLLECTIONS
```

Comment:
FX_ATLEAST_45 is not enough? [IReadonlyDictionary msdn](https://msdn.microsoft.com/en-us/library/hh136548%28v=vs.110%29.aspx)